ddl/syncer.go

```diff
@@ -380,3 +400,105 @@ func (s *schemaVersionSyncer) OwnerCheckAllVersions(ctx context.Context, latestV
 		time.Sleep(checkVersInterval)
 	}
 }
+
+const (
+	opDefaultRetryCnt = 10
+	failedGetTTLLimit = 20
+	opDefaultTimeout  = 3 * time.Second
+	opRetryInterval   = 500 * time.Millisecond
+	// NeededCleanTTL is exported for testing.
+	NeededCleanTTL = -60
+)
+
+func (s *schemaVersionSyncer) StartCleanWork() {
+	for {
+		time.Sleep(time.Second * 1)
+		if !s.ownerChecker.IsOwner() {
+			continue
+		}
+
+		select {
+		case <-s.notifyCleanExpiredPathsCh:
+			for i := 0; i < opDefaultRetryCnt; i++ {
```

Comment:
We can extract these codes into a func, to make it more readable.

Reply:
I  think it's OK. Here only get leases and do `doCleanExpirePaths`.

Reply:
but the `break` in line 433 is not very obvious.

Reply:
It does a retry, I think it's OK.